test/grouping.jl

```diff
@@ -2217,4 +2218,72 @@ end
     end
 end
 
+@testset "corner cases of group_reduce" begin
+    df = DataFrame(g=[1,1,1,2,2,2], x=Any[1,1,1,1.5,1.5,1.5])
+    gdf = groupby_checked(df, :g)
+    @test combine(groupby(df, :g), :x => sum) == DataFrame(g=1:2, x_sum=[3.0, 4.5])
```

Comment:
Also need to check the eltype.

Reply:
I will add the check, but please note that this check does not tell us what `groupreduce_init` returned, as the final eltype of the column is determined elsewhere.